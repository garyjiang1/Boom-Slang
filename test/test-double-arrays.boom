def print_darray(float[][] a) returns void:
	int i = 0
	int j = 0
	loop i+=1 while i < len(a):
		j = 0
		loop j+=1 while j < len(a[0]):
			println("arr[" + i + "][" + j + "] = " + a[i][j])

float[][] arr = default float[3][3]
arr[1][1] = 3.14
arr[2][0] = 4.2
print_darray(arr)
