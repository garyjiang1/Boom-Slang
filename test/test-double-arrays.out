arr[0][0] = 0.0000
arr[0][1] = 0.0000
arr[0][2] = 0.0000
arr[1][0] = 0.0000
arr[1][1] = 3.1400
arr[1][2] = 0.0000
arr[2][0] = 4.2000
arr[2][1] = 0.0000
arr[2][2] = 0.0000
